### ISSUE TITLE
startWorkflow not working

### ISSUE THREAD

**Issue by beruken** [Aymkdn/SharepointPlus#101]
I'm trying to fire off a sp2010 workflow from within SP2016. The 2013/16 WF manager is not yet enabled so we are using the 2010 default.  I found your status method and it returned this... 

"Possible Unhandled Promise Rejection: [SharepointPlus 'startWorkflow'] Please provide the item ID"

`$SP().list("Product Sheets", "http://mysite").startWorkflow({workflowName:"Create Product Sheet", 
					parameters:[{name:"TaskUID",value:kt.TaskId},{name:"TaskID",value:101},{name:"TaskName",value:kt.TaskName},{name:"Project",value:projectData.ProjectName},{name:"ProjectID",value:projectData.Id},{name:"EstDate",value:kt.TaskBaselineFinishDate},{name:"TaskStart",value:kt.TaskStartDate}]});
`
Any Idea what may be wrong?

**Comment by Aymkdn** [comment]
You have to provide the item ID. Please refer to the documentation.

**Comment by beruken** [comment]
Docs show optional, the item is not yet created. How can I use an ID. I did create this as a list workflow, however it really does not operate on list items. It creates one. 

I assume I need to re-create as a site workflow. Can I use parameters with the Site Workflows?

**Comment by Aymkdn** [comment]
Yes you have to create it as a site workflow if not related to an item, or use another existing item.

I think the parameters will also work with site workflows.

**Comment by beruken** [comment]
Since I have to select an item from the list to access the custom action button and initiate the WF I hardcoded an item ID and I got this.

Possible Unhandled Promise Rejection: TypeError: Object doesn't support property or method 'startsWith'

**Comment by Aymkdn** [comment]
Your error is not due to my code

**Comment by beruken** [comment]
ok. I'm not using Promise as that was problematic trying to wire it up. I do reference the polyfill.js as I am using IE11.

**Comment by Aymkdn** [comment]
Make sure to use the IE11 engine (with `<meta http-equiv="X-UA-Compatible" content="IE=Edge" />`), otherwise your page may run as IE7 and some functions like `String.startsWith` won't  work

**Comment by Aymkdn** [comment]
You can see with engine isso running by checking the web developer toolbar of IE

**Comment by beruken** [comment]
Thanks. Sorry for late response. I am referencing Edge. I also tried hard coded the values as well as the SPServices version.  That just reported back the workflow started. Nothing happened. I'm not understanding how to try this with a site workflow when your method is based on a List reference $SP().list(). Does the web service accept an empty value for the item property?

**Comment by Aymkdn** [comment]
Did you read the documentation? Because there is an example that I copy below : 
```
// if you want to call a Site Workflow, just leave the list name empty and don't provide an item ID, e.g.:
$SP().list("").startWorkflow({workflowName:"My Site Workflow"});
```

**Comment by beruken** [comment]
Yes I did. I got it working with the list version not site version of the workflow and SPServices. I still get the same Promise error "startsWith" with SPPLus. Fiddler showed it got far as executing the webservice and returning the itemUrl's list item from list.asmx but no other response.

The method you described resulted in this...

Possible Unhandled Promise Rejection: [SharepointPlus 'startWorkflow2013']: SP.SOD.executeFunc is required (from the Microsoft file called init.js)

Which makes sense because I am using a custom page with a custom master that has no references to the SP scripts. I am also not using 2013 workflows since they are not yet enabled. If I get a chance I will try to wrap it all up in a webpart page.

Thanks for your help.

**Comment by Aymkdn** [comment]
What is your code?

**Comment by beruken** [comment]
`$SP().list("").startWorkflow({workflowName:"Create` New Product", parameters:[{name:"TaskUID",value:"123465798"},{name:"TaskID",value:"101"},{name:"TaskName",value:"Fix EPMS"},{name:"Project",value:"0000"},{name:"ProjectID",value:"3"},{name:"EstDate",value:"2/3/2018"},{name:"TaskStart",value:"2/3/2018"}]});
return;
Promise.all([
	$SP().list("Product Sheets", "http://mysite").startWorkflow({ID:1178,workflowName:"Create Product Sheet", parameters:[{name:"TaskUID",value:"123465798"},{name:"TaskID",value:"101"},{name:"TaskName",value:"WTF new task"},{name:"Project",value:"0000"},{name:"ProjectID",value:"3"},{name:"EstDate",value:"2/3/2018"},{name:"TaskStart",value:"2/3/2018"}]})
     ]);
`

**Comment by Aymkdn** [comment]
When it's a site workflow the `startWorkflow2013` method is automatically called because it needs some code from there.  

This function needs `init.js` that is normally called on all Sharepoint pages. So you'll have to load init.js
Then 2 other Microsoft files are loaded : sp.js and sp.workflowservices.js
